### PR TITLE
Project start

### PR DESCRIPTION
**Pull request by IvanLuchkin**
- added .gitignore
- added all entities
- added Hibernate implementation of MovieDao
- added dependency injection tools
- added Hibernate configurations

### CODE REVIEW

**Comment by IvanLuchkin** [comment]
Please do not auto-approve or approve with requested changes

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            transaction = session.beginTransaction();
+            Long id = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(id);
```
Comment (Max-Pochepets) [inline src/main/java/cinema/dao/impl/MovieDaoImpl.java]: check one more [time](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html)

File: src/main/java/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
```
Comment (Max-Pochepets) [inline src/main/java/cinema/dao/impl/MovieDaoImpl.java]: same

File: src/main/java/cinema/model/Movie.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package cinema.model;
+
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+
+@Entity
+public class Movie {
```
Comment (Max-Pochepets) [inline src/main/java/cinema/model/Movie.java:10]: in the scope of this task only Movie should be present

File: src/main/java/cinema/util/HibernateUtil.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package cinema.util;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.cfg.Configuration;
+
+public class HibernateUtil {
+    private static final SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil() {
+
```
Comment (Max-Pochepets) [inline src/main/java/cinema/util/HibernateUtil.java]:
```suggestion
```

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/model/Movie.java (hunk 3 above)
Comment (IvanLuchkin) [reply src/main/java/cinema/model/Movie.java:10]: Well, I'm not going to remove these entities anyway...

**Review by Max-Pochepets — APPROVED** [review]
👍

File: src/main/java/cinema/model/CinemaHall.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package cinema.model;
+
+import java.util.Objects;
+
+public class CinemaHall {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/CinemaHall.java]: цього класу тут не повинно бути, це одна із наступних домашок

File: src/main/java/cinema/model/MovieSession.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
+package cinema.model;
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+public class MovieSession {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/MovieSession.java]: і цього

File: src/main/java/cinema/model/Order.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
+package cinema.model;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+
+public class Order {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/Order.java]: і цього

File: src/main/java/cinema/model/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
+package cinema.model;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ShoppingCart {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/ShoppingCart.java]: і цього

File: src/main/java/cinema/model/Ticket.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
+package cinema.model;
+
+import java.util.Objects;
+
+public class Ticket {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/Ticket.java]: the same

File: src/main/java/cinema/model/User.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
+package cinema.model;
+
+import java.util.Objects;
+
+public class User {
```
Comment (Sofasmile) [inline src/main/java/cinema/model/User.java]: )

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**